### ISSUE TITLE
*help* Evaluation of a model

### ISSUE THREAD

**Issue by nimaous** [tensorflow/tensor2tensor#744]
### Description
is there any standard method to evaluate a trained model?  I just replaced the test shards with new one and run `t2t-train` command with `--schedule = evaluate` and `--eval_run_autoregressive`.  is it a correct method of evaluating?

**Comment by martinpopel** [comment]
This is what I suggested you at Gitter because you were interested in the (Xentropy) loss and you had the model already trained and just wanted to test on new dataset. In this case, it is the only option. Note however that loss is an intrinsic and internal evaluation measure, hardly comparable across frameworks (or even different models within the same framework if they differ e.g. in vocabulary size). So be careful when comparing these numbers. (I guess most T2T users don't use the `--eval_run_autoregressive` flag, so their scores are not comparable to yours.)

**Comment by nimaous** [comment]
Thank you. what just made me a bit unsure about this process is that, when I use the same  test set with `--schedule = evaluate` and `--eval_run_autoregressive` I got accuracy of 1 and  without these flag set I got accuracy of 90.

**Comment by martinpopel** [comment]
Do you mean accuracy 0.01 vs. 0.90? This is really strange. What is your `--problem`? (BTW: the issue template suggests to report the full command used for training/evaluation.)

If it is machine translation, then `--schedule=evaluate` should have no effect (relative to the last evaluation in `continous_train_and_eval`) and `--eval_run_autoregressive` should not have so huge effect. So please, inspect which of the two options causes the difference.

If it is language modelling, then don't use `--eval_run_autoregressive` (and I am sorry for confusion) - it would mean generating the whole sentence without any clue what is the sentence about, which is clearly nonsense.

**Comment by nimaous** [comment]
Indeed by 1, I mean 100% accuracy. my problem is DNA translation which could be considered as a machine translation problem and I'm using transformer model. 
for the evaluation phas Im using following t2t-train command 
`t2t-trainer \
 --t2t_usr_dir=$USR_DIR  --schedule=evaluate  --eval_run_autoregressive   --eval_steps=1  
--data_dir=$DATA_DIR
 --problems=$PROBLEM 
 --model=$MODEL 
 --hparams_set=$HPARAMS 
 --worker_gpu=2 
 --output_dir=$TRAIN_DIR 
 --train_steps=0`

for the training-evaulation phase Im using following t2t-train command
`t2t-trainer 
 --t2t_usr_dir=$USR_DIR 
 --eval_steps=50 
 --data_dir=$DATA_DIR 
 --problems=$PROBLEM 
 --model=$MODEL 
 --hparams_set=$HPARAMS 
 --worker_gpu=2 
 --output_dir=$TRAIN_DIR 
 --train_steps=200000`

**Comment by martinpopel** [comment]
`--eval_steps=1` means that just the first batch (32 "sentences" by default) will be used for the evaluation. This is probably not what you want and may explain the 100% accuracy. (BTW: I understand "1" means 100%, but in that case I was confused by "accuracy of 90".) The default eval_steps is 100. To evaluate on the whole dev/test set, you can set eval_steps arbitrarily high, which will produce a warning after the data is exhausted, but you can ignore the warning.

If you are still unsure whether the accuracy is 90% or 100%, just use `t2t-decoder` and compare the output with the reference yourself.

**Comment by nimaous** [comment]
the problem was --eval_steps. I don't know why just put it 1 :). it is fixed now. tnx
